New update mechanism and other side changes

dein0s opened the pull request:
The initial plan is:
- add [EasyIni lib](https://github.com/dein0s/AHK_Snippets/blob/master/EasyIni.ahk)
- replace current update strategy for configs with the help of RIni (replace+backup -> update+backup)
- move map mods from txt to ini (easier to maintain)

Further actions:
- switch to EasyIni everywhere else (can take some time)

dein0s commented:
Any thoughts on this? Should I continue working on this?

Eruyome commented:
Sooner or later I'd have implemented something along those lines myself. So feel free to continue working on it.

dein0s commented:
Well, all things from the initial plan are implemented. Now I hate AHK more then before :D

Current changelist from commit message:
- EasyIni instead of RIni (TradeMacro and ItemInfo);
- MapModWarnings.txt -> MapModWarnings.ini (ItemInfo);
- AdditionalMacros.txt -> AdditionalMacros.ini + AdditionalMacros.txt (TradeMacro and ItemInfo);
- swapped key comments in config.ini (ItemInfo);
- added funcs to convert ^ files to the new format (TradeMacro and ItemInfo);
- added compare and update funcs (now user settings will persist after macro update, TradeMacro and ItemInfo)
- changed behaviour of PoEScripts_HandleUserSettings() to reflect all above changes (TradeMacro and ItemInfo)

Next big goal is to switch everywhere else from `IniRead/IniWrite` to `EasyIni`. I won't have that much spare time in a next few weeks, thus I'll make those changes in another PR.

As this changes affects both TradeMacro and ItemInfo, I think @aRTy42 also need to review them.

Feedback, questions, constructive criticism are welcome :)
